airflow/jobs.py

```diff
@@ -1701,6 +1701,16 @@ def _update_counters(self, started, succeeded, skipped, failed, tasks_to_run):
                                     .format(ti))
                 started.pop(key)
                 tasks_to_run[key] = ti
+            # special case: if we hit concurrency limits put it back in the queue
+            elif ti.state == State.NONE:
+                self.logger.warning("FIXME: task instance {} state was set to "
```

Comment:
I think I could be convinced either way, though for now a small comment about concurrency reached will help understand what this codepath.